api/src/main/java/io/minio/MinioClient.java

```diff
@@ -1387,6 +1388,127 @@ public void getObject(String bucketName, String objectName, String fileName)
     }
   }
 
+  /**
+   * Copy a source object into a new object with the provided name in the provided bucket.
+   *
+   * </p>
+   * <b>Example:</b><br>
+   *
+   * <pre>
+   * {@code minioClient.copyObject("my-bucketname", "my-objectname", "/my-sourcebucketname/my-objectsourcename");}
+   * </pre>
+   *
+   * @param destBucketName
+   *          Bucket name where the object will be created.
+   * @param destObjectName
+   *          Object name to be created after copy.
+   * @param sourceObjectPath
+   *          Path to the source object to be copied, is always
+   *          prefixed with a bucket.
+   *
+   * @throws InvalidBucketNameException
+   *           upon an invalid bucket name
+   * @throws NoSuchAlgorithmException
+   *           upon requested algorithm was not found during signature calculation
+   * @throws InvalidKeyException
+   *           upon an invalid access key or secret key
+   */
+  public void copyObject(String destBucketName, String destObjectName, String sourceObjectPath)
+      throws InvalidKeyException, InvalidBucketNameException, NoSuchAlgorithmException, InsufficientDataException,
+      NoResponseException, ErrorResponseException, InternalException, IOException, XmlPullParserException,
+      InvalidArgumentException {
+    copyObjectResult(destBucketName, destObjectName, sourceObjectPath, null);
+  }
+
+  /**
+   * Copy a source object into a new object with the provided name in the provided bucket.
+   * optionally can take a key value CopyConditions as well for conditionally attempting
+   * copyObject.
+   *
+   * </p>
+   * <b>Example:</b><br>
+   *
+   * <pre>
+   * {@code minioClient.copyObject("my-bucketname", "my-objectname", "/my-sourcebucketname/my-objectsourcename",
+   * copyConditions);}
+   * </pre>
+   *
+   * @param destBucketName
+   *          Bucket name where the object will be created.
```

Comment:
Can we add another line here saying `Bucket will be created if doesn't exist`? or it is implicit?